napari/layers/points/points.py

```diff
@@ -1721,6 +1729,8 @@ def _set_view_slice(self):
 
     def _make_slice_request(self, dims) -> _PointSliceRequest:
         """Make a Points slice request based on the given dims and these data."""
+        # indicate the layer is currently mid-load
+        self._loaded = False
```

Comment:
This approach means there can be times when `loaded` is `True` when it shouldn't be. For example:

```python
points = Points(...)
points.loaded  # => True, as expected
points._make_slice_request()  # 1st slice
points.loaded  # => False, as expected
points._make_slice_request()  # 2nd slice
points.loaded  # => False, as expected
points._update_slice_response(...)  # 1st slice
points.loaded  # => True, but should be False because the 2nd slice is not loaded
points._update_slice_response(...)  # 2nd slice
points.loaded  # => True, as expected
```

In order to always return the correct value, I think you need to know if there are other pending/active slice tasks for this layer. You could do that by storing an ID per request and checking that in `_update_slice_response`.

But I think a better solution is to update this state from `_LayerSlicer` which already has a way to check that state. That also has the benefit of making the state change more explicit rather than as a side-effect of `make_slice_request` where the intent was to making that a constant method (i.e. no state changes). And also doesn't add more slice state to the layer.

Reply:
You are absolutely correct. I lost sight of the multiple requests part of this. 

>In order to always return the correct value, I think you need to know if there are other pending/active slice tasks for this layer. You could do that by storing an ID per request and checking that in _update_slice_response.

Not sure if this can be done because the layer itself has no access to the slice tasks. Well I guess you could do it if you store the tasks on the layer itself, but that just adds more state.  

>But I think a better solution is to update this state from _LayerSlicer 

Exactly. My question is: how do we make this accessible from the layer? We already have a method `LayerSlicer._find_existing_task([Layer,])` which can discover tasks for any layer, but we don't have anything that can reach out from the layer itself. I'm wondering if that's even necessary to have?